src/Queue/SqsQueueAttributeDo.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace ShoppinPal\YapepCommon\Queue;
+
+class SqsQueueAttributeDo
+{
+
+    protected const ATTRIBUTE_MAP = [
+        'Policy'                                => 'policy',
+        'VisibilityTimeout'                     => 'visibilityTimeout',
+        'MaximumMessageSize'                    => 'maxiumMessageSize',
+        'MessageRetentionPeriod'                => 'maximumRetentionPeriod',
+        'ApproximateNumberOfMessages'           => 'approximateNumberOfMessages',
+        'ApproximateNumberOfMessagesNotVisible' => 'approximateNumberOfMessagesNotVisible',
+        'CreatedTimestamp'                      => 'createdTimestamp',
+        'LastModifiedTimestamp'                 => 'lastModifiedTimestamp',
+        'QueueArn'                              => 'queueArn',
+        'ApproximateNumberOfMessagesDelayed'    => 'approximateNumberOfMessagedDelayed',
+        'DelaySeconds'                          => 'delaySeconds',
+        'ReceiveMessageWaitTimeSeconds'         => 'receiveMessageWaitTimeSeconds',
+        'RedrivePolicy'                         => 'redrivePolicy',
+        'FifoQueue'                             => 'fifoQueue',
+        'ContentBasedDeduplication'             => 'contentBasedDeduplication',
+        'KmsMasterKeyId'                        => 'kmsMasterKeyId',
+        'KmsDataKeyReusePeriodSeconds'          => 'kmsDataKeyReusePeriodSeconds',
+    ];
+
+    /** @var int|null */
+    protected $approximateNumberOfMessagedDelayed;
+
+    /** @var int|null */
+    protected $approximateNumberOfMessages;
+
+    /** @var int|null */
+    protected $approximateNumberOfMessagesNotVisible;
+
+    /** @var bool|null */
+    protected $contentBasedDeduplication;
+
+    /** @var string|null */
+    protected $createdTimestamp;
+
+    /** @var int|null */
+    protected $delaySeconds;
+
+    /** @var bool|null */
+    protected $fifoQueue;
+
+    /** @var string|null */
+    protected $kmsDataKeyReusePeriodSeconds;
+
+    /** @var string|null */
+    protected $kmsMasterKeyId;
+
+    /** @var int|null */
+    protected $lastModifiedTimestamp;
+
+    /** @var int|null */
+    protected $maximumRetentionPeriod;
+
+    /** @var int|null */
+    protected $maxiumMessageSize;
+
+    /** @var string|null */
+    protected $policy;
+
+    /** @var string|null */
+    protected $queueArn;
+
+    /** @var int|null */
+    protected $receiveMessageWaitTimeSeconds;
+
+    /** @var string|null */
+    protected $redrivePolicy;
+
+    /** @var int|null */
+    protected $visibilityTimeout;
+
+    public function __construct(array $attributes)
+    {
+        foreach (self::ATTRIBUTE_MAP as $attributeName => $propertyName) {
+            $this->$propertyName = $attributes[$attributeName] ?? null;
```

Comment:
Are you sure that the input array will have the proper types?

Reply:
This is coming from AWS, so we are not in control of the types really. But I'm intentionally not using strict types here to let PHP coerce to the correct types.